core/src/main/java/net/opentsdb/query/processor/groupby/GroupBy.java

```diff
@@ -44,9 +54,13 @@
  * @since 3.0
  */
 public class GroupBy extends AbstractQueryNode {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractQueryNode.class);
   
   /** The config for this group by node. */
   private final GroupByConfig config;
+
+  private DownsampleConfig downsampleConfig;
```

Comment:
java doc it as `An optional downsample config for use by the GroupByArrayIterator to size it's array properly when running in parallel.`